File: 0005-smallest_multiple/main.py
#!/usr/bin/env python


def greatest_common_divisor(a: int, b: int) -> int:
    while b != 0:
        a, b = b, a % b

    return a


def lowest_common_multiplier(a: int, b: int) -> int:
    return abs(a * b) // greatest_common_divisor(a, b)


def main():
    limit = 20
    sequence = list(range(1, limit))
    result = sequence[0]

    for number in sequence[1:]:
        result = lowest_common_multiplier(result, number)

    print(result)


if __name__ == "__main__":
    main()

File: 0005-smallest_multiple/main.rs
fn gcd(mut a: i64, mut b: i64) -> i64 {
    while b != 0 {
        let tmp = b;
        b = a % b;
        a = tmp;
    }
    
    a
}

fn lcm(a: i64, b: i64) -> i64 {
    (a * b).abs() / gcd(a, b)
}

fn main() {
    let limit: i64 = 10;
    let mut sequence: Vec<i64> = (1..limit).collect();
    let mut result: i64 = sequence[0];
    
    for number in &mut sequence[1..] {
        result = lcm(result, *number);
    }
    
    println!("{}", result);
}
